File: frontend/src/types/time-entry.ts
export interface TimeEntry {
  id: number
  projectId: number
  projectTitle: string
  userId: number
  userName: string
  minutes: number
  workDate: string
  description: string | null
  billable: boolean
  rateCents: number | null
  createdAt: string
  updatedAt: string
}

export interface TimeEntryInput {
  projectId: number
  minutes: number
  workDate: string
  description?: string | null
  billable?: boolean
  rateCents?: number | null
}

export interface TimeEntrySummary {
  totalMinutes: number
  billableMinutes: number
  totalRevenueCents: number
}

File: backend/src/main/java/com/ukizuru/consulting/models/TimeEntry.java
package com.ukizuru.consulting.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDate;
import java.util.Objects;

@Entity
@Table(name = "time_entries")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString(of = {"id", "minutes", "workDate"})
public class TimeEntry extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "project_id", nullable = false)
    private Project project;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @NotNull
    @Min(1)
    @Column(nullable = false)
    private Integer minutes;

    @NotNull
    @Column(name = "work_date", nullable = false)
    private LocalDate workDate;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private Boolean billable;

    @Column(name = "rate_cents")
    private Integer rateCents;

    @PrePersist
    void onPrePersist() {
        if (billable == null) billable = true;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof TimeEntry other)) return false;
        return id != null && id.equals(other.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}
